phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexUpgradeTool.java

```diff
@@ -418,7 +418,9 @@ private void rebuildIndexes(Connection conn, Configuration conf, String dataTabl
                 indexingTool = new IndexTool();
             }
             indexingTool.setConf(conf);
-            rebuildIndexes(conn, dataTableFullName, indexingTool);
+            if(rebuildMap.get(dataTableFullName).size() != 0 ) {
```

Comment:
I believe the crash was since rebuildMap didn't have anything for dataTableFullName. This change only moves the crash point.

Reply:
done. Used the file you shared as a starting point.